storage/block/result.go

```diff
@@ -216,8 +223,19 @@ func (it *filteredBlocksMetadataIter) Next() bool {
 	}
 	it.id = it.res[it.resIdx].ID
 	block := blocks[it.blockIdx]
-	// TODO(r): When reviving the end to end repair process, propagate tags.
-	it.metadata = NewMetadata(it.id, nil, block.Start,
+	var tags ident.Tags
+	if tagsIter := it.res[it.resIdx].Tags; tagsIter != nil {
+		for tagsIter.Next() {
+			curr := tagsIter.Current()
+			tags.Append(ident.StringTag(curr.Name.String(), curr.Value.String()))
+		}
+		if err := tagsIter.Err(); err != nil {
+			it.err = err
+			return false
+		}
+		tagsIter.Close()
```

Comment:
do you need to put this in the early terminate return too?

Reply:
I'm fine with leaking things in the extreme rare error case if it makes the code more simple.

Reply:
fair enough